poly/src/polynomial/mod.rs

```diff
@@ -57,6 +58,16 @@ pub trait UVPolynomial<F: Field>: Polynomial<F, Point = F> {
 
 /// Describes the interface for univariate polynomials
 pub trait MVPolynomial<F: Field>: Polynomial<F> {
+    /// Returns the number of variables in `self`
+    fn num_vars(&self) -> usize;
+
+    /// Outputs an `l`-variate polynomial which is the sum of `l` `d`-degree univariate
+    /// polynomials where each coefficient is sampled uniformly at random.
+    fn rand<R: Rng>(d: usize, num_vars: usize, rng: &mut R) -> Self;
```

Comment:
Not really sure that it makes sense to have rand on the `MVPolynomial` trait, since the function doesn't really make sense in its current form for sparse polynomials, as you need more info to specify how sparse it is. (Ditto for `UVPolynomial`)

Maybe we should remove it from the trait here and in `UVPolynomial`? Thoughts @Pratyush @ryanleh 
(This can be done in a separate PR for removing from `UVPolynomial`)